syft/pkg/cataloger/golang/licenses.go

```diff
@@ -31,6 +32,7 @@ type goLicenses struct {
 	opts                  GoCatalogerOpts
 	localModCacheResolver file.WritableResolver
 	progress              *monitor.CatalogerTask
+	licenseFileNames      *strset.Set
```

Comment:
Can you help me understand why this is needed? There wasn't a string set here before.

Reply:
this is removing the string set that was in static space, which was previously assignable (thus not a good long term approach). While removing the internal string set usage I did not elect to replace it on the API in the same way, instead removing it's usage entirely. The place it was used for convenience was in this object, so that's why it was added here.